### PR TITLE
[BEAM-12686] Enable self-managed relative timers

### PR DESCRIPTION
**Pull request by reuvenlax**

### CODE REVIEW

**Comment by chamikaramj** [comment]
Is this worth an email to the dev list? 
(I'm not sure if this is a corner case or a significant change of behavior)

**Comment by chamikaramj** [comment]
LGTM

File: sdks/java/core/src/main/java/org/apache/beam/sdk/state/Timer.java [hunk 1]
```diff
@@ -98,4 +98,11 @@
    * the timer fires.
    */
   Timer withOutputTimestamp(Instant outputTime);
+
+  /**
+   * Returns the current relative time use by {@link #setRelative()} and {@link #offset}. This can
```
Comment (chamikaramj) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/state/Timer.java:103]: s/use/used

**Comment by reuvenlax** [comment]
run flink validatesrunner

**Comment by reuvenlax** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Comment by reuvenlax** [comment]
Run Samza ValidatesRunner

**Comment by reuvenlax** [comment]
Run Spark ValidatesRunner

**Comment by reuvenlax** [comment]
Run Java Spark PortableValidatesRunner Batch

**Comment by reuvenlax** [comment]
Run Twister2 ValidatesRunner

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
run flink validatesrunner

**Comment by reuvenlax** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Pull request merged**